### PR TITLE
Add TypeScript definition files

### PR DESCRIPTION
**Pull request by littlepiggy03**
You can use following code to check their correctness.

```shell
npm i -g dtslint
npm run dtslint
```

In `types/index.d.ts`,
```typescript
import * as DRange from "drange";
// Like `const DRange = require("drange");`

declare class RandExp {
    static sugar(): void;
    // Is a static function
    
    constructor(pattern: string | RegExp, flags?: string);
    // `string | RegExp` means "`string` or `RegExp`"

    max: number
    // Is a `number` member variable
    
    randInt: (from: number, to: number) => number;
    // Is a member variable of a function which takes two `number` and returns a `number`
}
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -128,3 +128,18 @@ Tests are written with [mocha](https://mochajs.org)
...
 ```
+
+# Integration with TypeScript
+
+DRange includes TypeScript definitions.
+
+```typescript
+import * as RandExp from "randexp";
+const randExp = new RandExp(/[a-z]{6}/);
+RandExp.gen();
```
Comment (fent) [inline README.md]: Typo

File: README.md [hunk 2]
```diff
@@ -128,3 +128,18 @@ Tests are written with [mocha](https://mochajs.org)
 ```bash
 npm test
 ```
+
+# Integration with TypeScript
+
+DRange includes TypeScript definitions.
```
Comment (fent) [inline README.md]: RandExp

File: README.md [hunk 3]
```diff
@@ -128,3 +128,18 @@ Tests are written with [mocha](https://mochajs.org)
...
+
+```typescript
+import * as RandExp from "randexp";
+const randExp = new RandExp(/[a-z]{6}/);
```
Comment (fent) [inline README.md]: Would prefer all lowercased variable to have it consistent with the rest of the examples on the readme

File: package.json [hunk 4]
```diff
@@ -17,17 +17,19 @@
...
   "files": [
-    "lib"
+    "lib",
+    "types/index.d.ts"
   ],
   "scripts": {
     "version": "gulp build && git add build",
-    "test": "istanbul cover node_modules/.bin/_mocha -- test/*-test.js"
+    "test": "istanbul cover node_modules/.bin/_mocha -- test/*-test.js",
+    "dtslint": "dtslint types --onlyTestTsNext"
```
Comment (fent) [inline package.json]: I'm not too familiar with dtslint, why this option?

File: types/index.d.ts [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+declare namespace RandExp {}
+
+declare class RandExp {
+    static randexp(pattern: string | RegExp, flags?: string): string;
+    static sugar(): void;
+    constructor(pattern: string | RegExp, flags?: string);
+    gen(): string;
+    defaultRange: DRange;
+    randInt: (from: number, to: number) => number;
+    max: number;
```
Comment (fent) [inline types/index.d.ts]: Not sure about the inconsistent indent in these files. Some use 4 spaces, others 2, the project uses 2. Is using 4 the default in typescript?

Comment (littlepiggy03) [reply types/index.d.ts]: In TypeScript, 4 spaces is used in docs/generated codes. Using 4 seems to be default in TypeScript files.

**Comment by littlepiggy03** [comment]
@fent Are you still watching this PR? I'm going to move this PR to [DefinitelyTyped](https://github.com/DefinitelyTyped/DefinitelyTyped).

**Comment by fent** [comment]
Yep, I can look at this again. But which is preferred? For the definition files to exist here or in DefinitelyTyped?

**Comment by littlepiggy03** [comment]
I'd like to add definition file here to smaller my package. Could you review my changes again?

**Comment by fent** [comment]
Looks good. Thank you! 👍

**Comment by littlepiggy03** [comment]
Thank for you too, this PR helps me a lot.

**Pull request merged**